tracer/test/Datadog.Trace.Tests/Telemetry/TelemetrySettingsTests.cs

```diff
@@ -134,9 +133,72 @@ public void SetsTelemetryEnabledBasedOnApiKeyAndEnabledSettings(string apiKey, s
                 { ConfigurationKeys.ApiKey, apiKey },
             });
 
-            var settings = new TelemetrySettings(source, _tracerSettings);
+            var settings = TelemetrySettings.FromSource(source);
+            if (enabled && !string.IsNullOrEmpty(apiKey))
```

Comment:
Did you mean to use `settings.TelemetryEnabled`? It feels weird to use an output/expectation to control the if/branch

Reply:
I dunno, this feels _right_ to me 🤔 It's basically a "calculated" another parameter to the method, which is dependent on the inputs. using `settings.TelemetryEnabled` means we could test the wrong branch if the logic in `FromSource` is wrong. The method would still _fail_ but it might be confusing.

Though, to be fair, it's all a bit confusing atm 😅 I extracted as a variable to try and make that clearer in https://github.com/DataDog/dd-trace-dotnet/pull/2617/commits/268f3caa33852467feec3a28e3e528b0011dfa40